src/orchestrator.ts

```diff
@@ -595,16 +595,29 @@ export class Orchestrator {
         state: HistoryEvent[],
         name: string,
         instanceId: string)
-        : SubOrchestrationInstanceCreatedEvent {
-        const returnValue = name
-            ? state.filter((val: HistoryEvent) => {
+        : SubOrchestrationInstanceCreatedEvent | undefined {
+
+        if (name) {
```

Comment:
I misread this logic originally. You can improve code readability and reduce complexity if you do the error checking in the `if` block and then leave out the `else` block. Something like this:

```javascript
if (!name) {
    throw new Error("...");
}

const matches = state.filter(...
```

Reply:
Null checking made this irrelevant :)